spec/integration/puppet_x/puppetlabs/scheduled_task/taskscheduler2_task_spec.rb

```diff
@@ -0,0 +1,190 @@
+
+#!/usr/bin/env ruby
+require 'spec_helper'
+
+require 'puppet_x/puppetlabs/scheduled_task/taskscheduler2_task' if Puppet.features.microsoft_windows?
+
+RSpec::Matchers.define :be_same_as_powershell_command do |ps_cmd|
+  define_method :run_ps do |cmd|
+    full_cmd = "powershell.exe -NoLogo -NoProfile -NonInteractive -Command \"#{cmd}\""
+
+    result = `#{full_cmd}`
+
+    result.strip
+  end
+
+  match do |actual|
+    from_ps = run_ps(ps_cmd)
+
+    # This matcher probably won't tolerate UTF8 characters
+    actual.to_s == from_ps
+  end
+
+  failure_message do |actual|
+    "expected that #{actual} would match #{run_ps(ps_cmd)} from PowerShell command #{ps_cmd}"
+  end
+end
+
+def triggers
+  now = Time.now
+
+  defaults = {
+    'end_day'                 => 0,
+    'end_year'                => 0,
+    'minutes_interval'        => 0,
+    'end_month'               => 0,
+    'minutes_duration'        => 0,
+    'start_year'              => now.year,
+    'start_month'             => now.month,
+    'start_day'               => now.day,
+    'start_hour'              => now.hour,
+    'start_minute'            => now.min,
+  }
+
+  [
+    # dummy time trigger
+    defaults.merge({
+      'random_minutes_interval' => 1,
+      'trigger_type'            => :TASK_TIME_TRIGGER_ONCE,
+    }),
+    defaults.merge({
+      'type'                    => { 'days_interval' => 1 },
+      'trigger_type'            => :TASK_TIME_TRIGGER_DAILY,
+    }),
+    defaults.merge({
+      'type'         => {
+        'weeks_interval' => 1,
+        'days_of_week'   => Win32::TaskScheduler::MONDAY,
+      },
+      'trigger_type'            => :TASK_TIME_TRIGGER_WEEKLY,
+    }),
+    defaults.merge({
+      'type'         => {
+        'months' => Win32::TaskScheduler::JANUARY,
+        # TODO: this value seems to be cooked up wrong
```

Comment:
We still have this test TODO here which should probably be removed / sorted out...